### PR TITLE
Simplify POST API requests for data entry

### LINKED ISSUE #261: Simplify POST API requests in frontend
#256 simplifies behaviour for `DELETE` API requests using the frontend API client to use a simple async callback instead of `useEffect` hooks because that pattern is cumbersome and [not recommended](https://react.dev/learn/you-might-not-need-an-effect#sending-a-post-request).

We should refactor our `POST` API calls to be similar to what is implemented for `DELETE` requests in #256. We can pick this up after #133 to prevent conflicts.

### PR DESCRIPTION
**Pull request by praseodym**
Closes #261

Required to implement the save functionality in #212.

### CODE REVIEW

File: frontend/app/component/form/candidates_votes_form/CandidatesVotesForm.tsx [hunk 1]
```diff
@@ -39,6 +40,7 @@ export interface CandidatesVotesFormProps {
 export function CandidatesVotesForm({ group }: CandidatesVotesFormProps) {
   const { register, format, deformat, warnings: inputMaskWarnings } = usePositiveNumberInputMask();
   const formRef = React.useRef<CandidatesVotesFormElement>(null);
+  const [saving, setSaving] = useState(false);
```
Comment (Lionqueen94) [inline frontend/app/component/form/candidates_votes_form/CandidatesVotesForm.tsx:43]:
```suggestion
  const [saving, setSaving] = React.useState(false);
```

File: frontend/app/component/form/candidates_votes_form/CandidatesVotesForm.tsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import * as React from "react";
+import { useState } from "react";
```
Comment (Lionqueen94) [inline frontend/app/component/form/candidates_votes_form/CandidatesVotesForm.tsx:2]:
```suggestion
```

File: frontend/app/component/form/differences/DifferencesForm.tsx [hunk 3]
```diff
@@ -41,6 +42,7 @@ export function DifferencesForm() {
   } = usePositiveNumberInputMask();
   const formRef = React.useRef<DifferencesFormElement>(null);
   usePreventFormEnterSubmit(formRef);
+  const [saving, setSaving] = useState(false);
```
Comment (Lionqueen94) [inline frontend/app/component/form/differences/DifferencesForm.tsx:45]:
```suggestion
  const [saving, setSaving] = React.useState(false);
```

File: frontend/app/component/form/differences/DifferencesForm.tsx [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import * as React from "react";
+import { useState } from "react";
```
Comment (Lionqueen94) [inline frontend/app/component/form/differences/DifferencesForm.tsx:2]:
```suggestion
```

File: frontend/app/component/form/recounted/RecountedForm.tsx [hunk 5]
```diff
@@ -17,6 +18,7 @@ export function RecountedForm() {
   const [hasValidationError, setHasValidationError] = React.useState(false);
   const formRef = React.useRef<RecountedFormElement>(null);
   usePreventFormEnterSubmit(formRef);
+  const [saving, setSaving] = useState(false);
```
Comment (Lionqueen94) [inline frontend/app/component/form/recounted/RecountedForm.tsx:21]:
```suggestion
  const [saving, setSaving] = React.useState(false);
```

File: frontend/app/component/form/recounted/RecountedForm.tsx [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import * as React from "react";
+import { useState } from "react";
```
Comment (Lionqueen94) [inline frontend/app/component/form/recounted/RecountedForm.tsx:2]:
```suggestion
```

File: frontend/app/component/form/voters_and_votes/VotersAndVotesForm.tsx [hunk 7]
```diff
@@ -46,6 +47,7 @@ export function VotersAndVotesForm() {
   } = usePositiveNumberInputMask();
   const formRef = React.useRef<VotersAndVotesFormElement>(null);
   usePreventFormEnterSubmit(formRef);
+  const [saving, setSaving] = useState(false);
```
Comment (Lionqueen94) [inline frontend/app/component/form/voters_and_votes/VotersAndVotesForm.tsx:50]:
```suggestion
  const [saving, setSaving] = React.useState(false);
```

File: frontend/app/component/form/voters_and_votes/VotersAndVotesForm.tsx [hunk 8]
```diff
@@ -1,4 +1,5 @@
 import * as React from "react";
+import { useState } from "react";
```
Comment (Lionqueen94) [inline frontend/app/component/form/voters_and_votes/VotersAndVotesForm.tsx:2]:
```suggestion
```

File: frontend/lib/api/form/pollingstation/PollingStationFormController.tsx [hunk 9]
```diff
@@ -375,89 +296,144 @@ export function PollingStationFormController({
...
-                  voters_recounts = old.voters_recounts;
-                }
+  const submitCurrentForm = async (ignoreWarnings?: boolean) => {
+    // React state is fixed within one render, so we update our own copy instead of using setValues directly
+    let newValues: PollingStationValues = values;
+    if (currentForm.current) {
+      const ref: AnyFormReference = currentForm.current;
+
+      //flag this submit to ignore warnings
+      _ignoreWarnings.current = ignoreWarnings ? ref.id : null;
```
Comment (lkleuver) [inline frontend/lib/api/form/pollingstation/PollingStationFormController.tsx:306]: This flag probably isn't necessary anymore ?

File: frontend/lib/api/form/pollingstation/PollingStationFormController.tsx [hunk 10]
```diff
@@ -466,12 +442,9 @@ export function PollingStationFormController({
   return (
     <PollingStationControllerContext.Provider
       value={{
+        error,
         formState,
         values,
-        setValues,
-        loading,
```
Comment (lkleuver) [inline frontend/lib/api/form/pollingstation/PollingStationFormController.tsx:472]: unsure about a local loading state while the entire form (all sections) are in that same state. 
Can you block navigation while loading for instance?

**Review by lkleuver — DISMISSED** [review]

File: frontend/lib/api/form/pollingstation/PollingStationFormController.tsx (hunk 9 above)
Comment (praseodym) [reply frontend/lib/api/form/pollingstation/PollingStationFormController.tsx]: Indeed, removed in b8f78c45e214f4ea949d25b7387fbf1b20da2d79.

File: frontend/lib/api/form/pollingstation/PollingStationFormController.tsx (hunk 10 above)
Comment (praseodym) [reply frontend/lib/api/form/pollingstation/PollingStationFormController.tsx:472]: I've moved it back up to the controller in a58728d0c5a46bd293e785118f79f565bd2e6745. Indeed useful when we want to block navigation.

**Review by Lionqueen94 — APPROVED** [review]

**Review by lkleuver — APPROVED** [review]

**Pull request merged**